native/src/main/java/io/ballerina/lib/ldap/Utils.java

```diff
@@ -80,19 +81,17 @@ public static BError createError(String message, Throwable throwable) {
     }
 
     private static BMap<BString, Object> getErrorDetails(Throwable throwable) {
-        if (throwable instanceof LDAPException) {
-            String resultCode = ((LDAPException) throwable).getResultCode().getName().toUpperCase();
-            String message = throwable.getMessage();
-            return ValueCreator.createRecordValue(getModule(), ERROR_DETAILS,
-                                                  Map.of(RESULT_STATUS, resultCode, ERROR_MESSAGE, message));
+        if (throwable instanceof LDAPException ldapException) {
+            String resultCode = ldapException.getResultCode().getName().toUpperCase(Locale.ROOT);
+            return ValueCreator.createRecordValue(getModule(), ERROR_DETAILS, Map.of(RESULT_STATUS, resultCode));
         }
         return ValueCreator.createRecordValue(getModule(), ERROR_DETAILS);
     }
```

Comment:
Instead of using an instanceof check, I'd suggest using method overloading here, which is more Java-ish way for handling these scenarios.